NebulaWorld/Factory/FactoryManager.cs

```diff
@@ -34,6 +41,114 @@ public static void Initialize()
             TargetPlanet = PLANET_NONE;
         }
 
+        public static void AddPlanetTimer(int planetId)
+        {
+            using (GetPlanetTimers(out var planetTimers))
+            {
+                // We don't want to load or unload the planet we are currently on
+                if (planetId == GameMain.localPlanet?.id)
+                {
+                    return;
+                }
+
+                // .NET Framework 4.7.2 does not have TryAdd so we must make sure the dictionary does not already contain the planet
+                if (!planetTimers.ContainsKey(planetId))
+                {
+                    // We haven't loaded this planet, let's load it
+                    LoadPlanetData(planetId);
+
+                    // Create a new 10 second timer for this planet
+                    planetTimers.Add(planetId, new Timer(TimeSpan.FromSeconds(10).TotalMilliseconds));
```

Comment:
What stops the timers? This timer should be AutoReset=false or something otherwise this will keep triggering every 10 seconds and uses a threadpool thread to do it.

Reply:
When the timer elapses it calls PlanetTimer_Elapsed(int planetId) which calls RemovePlanetTimer(planetId) which stops the timer and sets it to null

Reply:
Setting it to null won't stop the timer :p

Reply:
> This type implements the IDisposable interface. When you have finished using the type, you should dispose of it either directly or indirectly. To dispose of the type directly, call its Dispose method in a try/catch block. To dispose of it indirectly, use a language construct such as using (in C#) or Using (in Visual Basic). For more information, see the "Using an Object that Implements IDisposable" section in the IDisposable interface topic.

Reply:
It's possible that the Timer implementation uses weak keys to store the timer, but Stop only stops the timer from being fired, not from being added to the global timer queue.

Reply:
Oh, it's wrapped in a using statement there, I missed it in my initial review. It's fine.

Reply:
Hrm so the using statement is wrapping the Locker GetPlanetTimers() object, would the timer itself also be disposed after this? I copied the GetPrebuildRequests implementation when I did this so I don't fully understand how it works tbh, maybe disposal is required 😅